File: frontend/src/types/advisorPackets.ts
export interface AdvisorPacketWarning {
  code: string;
  message: string;
}

export interface AdvisorPacketRedaction {
  data_class: string;
  default_visibility: string;
  status: "included" | "excluded" | "summary_only";
  reason: string;
}

export interface AdvisorPacketOpportunitySummary {
  id: string;
  workspace_id: string;
  title: string;
  company_name: string;
  status: string;
  location: string | null;
  remote_type: string | null;
}

export interface AdvisorPacketApplicationSummary {
  id: string;
  current_state: string;
  applied_at: string | null;
  next_action_at: string | null;
  counts: Record<string, number>;
}

export interface AdvisorPacketArtifactSummary {
  id: string;
  artifact_type: string;
  title: string;
  lifecycle_status: string | null;
  revision_number: number | null;
  content_included: boolean;
  updated_at: string;
  warnings: AdvisorPacketWarning[];
}

export interface AdvisorPacket {
  packet_version: string;
  mode: "local_preview";
  generated_at: string;
  local_only: boolean;
  external_sharing_enabled: boolean;
  advisory_notice: string;
  opportunity: AdvisorPacketOpportunitySummary;
  application: AdvisorPacketApplicationSummary;
  artifacts: AdvisorPacketArtifactSummary[];
  redactions: AdvisorPacketRedaction[];
  warnings: AdvisorPacketWarning[];
}

File: backend/app/schemas/advisor_packets.py
from datetime import datetime
from typing import Literal
import uuid

from pydantic import BaseModel, Field


class AdvisorPacketWarning(BaseModel):
    code: str
    message: str


class AdvisorPacketRedaction(BaseModel):
    data_class: str
    default_visibility: str
    status: Literal["included", "excluded", "summary_only"]
    reason: str


class AdvisorPacketOpportunitySummary(BaseModel):
    id: uuid.UUID
    workspace_id: uuid.UUID
    title: str
    company_name: str
    status: str
    location: str | None = None
    remote_type: str | None = None


class AdvisorPacketApplicationSummary(BaseModel):
    id: uuid.UUID
    current_state: str
    applied_at: datetime | None = None
    next_action_at: datetime | None = None
    counts: dict[str, int] = Field(default_factory=dict)


class AdvisorPacketArtifactSummary(BaseModel):
    id: uuid.UUID
    artifact_type: str
    title: str
    lifecycle_status: str | None = None
    revision_number: int | None = None
    content_included: bool = False
    updated_at: datetime
    warnings: list[AdvisorPacketWarning] = Field(default_factory=list)


class AdvisorPacketResponse(BaseModel):
    packet_version: str
    mode: Literal["local_preview"]
    generated_at: datetime
    local_only: bool
    external_sharing_enabled: bool
    advisory_notice: str
    opportunity: AdvisorPacketOpportunitySummary
    application: AdvisorPacketApplicationSummary
    artifacts: list[AdvisorPacketArtifactSummary] = Field(default_factory=list)
    redactions: list[AdvisorPacketRedaction] = Field(default_factory=list)
    warnings: list[AdvisorPacketWarning] = Field(default_factory=list)
